clients/js/test/plugins/asset/addBlocker.test.ts

```diff
@@ -0,0 +1,177 @@
+import test from 'ava';
+import { generateSigner } from '@metaplex-foundation/umi';
+
+import { addPluginV1, createPlugin, pluginAuthorityPair } from '../../../src';
+import {
+  DEFAULT_ASSET,
+  assertAsset,
+  createAsset,
+  createUmi,
+} from '../../_setup';
+
+test('it cannot add UA-managed plugin if addBlocker had been added on creation', async (t) => {
+  // Given a Umi instance and a new signer.
+  const umi = await createUmi();
+
+  const asset = await createAsset(umi, {
+    plugins: [
+      pluginAuthorityPair({
+        type: 'AddBlocker',
+      }),
+    ],
+  });
+
+  const result = addPluginV1(umi, {
+    asset: asset.publicKey,
+    plugin: createPlugin({
+      type: 'PermanentFreezeDelegate',
+      data: { frozen: true },
+    }),
+  }).sendAndConfirm(umi);
```

Comment:
I think you should use a different test plugin here because permanent freeze delegate can never be added after creation anyways.